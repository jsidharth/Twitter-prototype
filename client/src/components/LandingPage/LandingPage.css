.flexImage {
  display: flex;
  flex-direction: row;
}
.btn-info-solid {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 15px;
  color: white;
  font-weight: 900;
  line-height: 15px;
  border: 2px solid #00acee;
  border-radius: 40px;
  transition: all 0.3s ease 0s;
  background-color: #00acee;
  height: 60%;
  width: 100%;
  margin-bottom: 10%;
}
.btn-info-transparent {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 15px;
  color: #00acee;
  font-weight: 900;
  line-height: 15px;
  border: 2px solid #00acee;
  border-radius: 40px;
  background: transparent;
  transition: all 0.3s ease 0s;
  height: 150%;
  width: 100%;
}
.btn-info-transparent-login {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 25px;
  color: #00acee;
  letter-spacing: 1px;
  line-height: 15px;
  border: 1px solid #00acee;
  border-radius: 40px;
  background: transparent;
  transition: all 0.3s ease 0s;
  height: 50%;
  width: 100%;
  margin-top: 25%;
  margin-left: 30%;
}
.info-transparent-login {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 15px;
  font-weight: 900;
  color: #00acee;
  line-height: 15px;
  border: 1px solid #00acee;
  border-radius: 40px;
  background: transparent;
  transition: all 0.3s ease 0s;
  height: 42%;
  width: 70%;
  margin-top: 40%;
  margin-left: 10%;
}

.container {
  margin-left: 0px;
  margin-right: 0px;
}
.signupLogin {
  margin-top: 500%;
}
.main1 {
  height: 100vh;
}

.home {
  background-image: url('../Image/twitterImage.png');
  background-size: 104%;
  background-repeat: repeat-y;
  width: 50%;
}

.modal {
  top: 10%;
  left: 10%;
}
.modal-header {
  border-bottom-color: white !important;
}
.modal-footer {
  border-top-color: white !important;
  justify-content: center !important;
  margin-top: 0% !important;
}

.btn-info-solid-next {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: bolder;
  font-size: 15px;
  color: white;
  letter-spacing: 0px;
  padding-bottom: 5px;
  line-height: 15px;
  border: 2px solid #00acee;
  border-radius: 40px;
  transition: all 0.3s ease 0s;
  background-color: #00acee;
  height: 30px;
  width: 100%;
}

.form-group {
  margin-bottom: 40px;
  outline: 0px;
}
.form-control-label {
  font-family: Arial;
  font-size: 10px;
  color: #6c6c6c;
  font-weight: bold;
  letter-spacing: 1px;
}

.login-username {
  border: none;
  background-color: #f0f0f5;
  border-bottom: 2px solid darkgray;
  border-top: 0px;
  border-radius: 0px;
  font-weight: bold;
  outline: 0;
  color: black;
  padding-left: 0px;
  height: 45px;
  width: 100% !important;
  margin-left: 10%;
}
.login-label {
  margin-top: 5%;
  margin-left: 10%;
}
input[type='text'] {
  border: none;
  background-color: #f0f0f5;
  border-bottom: 2px solid darkgray;
  border-top: 0px;
  border-radius: 0px;
  outline: 0;
  color: black;
  padding-left: 0px;
  height: 45px;
  width: 70%;
}
input[type='password'] {
  border: none;
  background-color: #f0f0f5;
  border-bottom: 2px solid darkgray;
  border-top: 0px;
  border-radius: 0px;
  outline: 0;
  color: black;
  padding-left: 0px;
  height: 45px;
  width: 70%;
}
input[type='text']:focus {
  border-bottom: 2px solid #00acee;
  box-shadow: none;
}
input[type='password']:focus {
  border-bottom: 2px solid #00acee;
  box-shadow: none;
}
input[type='email']:focus {
  border-bottom: 2px solid #00acee;
  box-shadow: none;
}
input[type='text']:read-only {
  border-bottom: 2px solid darkgray;
  box-shadow: none;
}
input[type='password']:read-only {
  border-bottom: 2px solid darkgray;
  box-shadow: none;
}
input[type='email'] {
  border: none;
  background-color: #f0f0f5;
  border-bottom: 2px solid darkgray;
  border-top: 0px;
  border-radius: 0px;
  outline: 0;
  color: black;
  padding-left: 0px;
  height: 45px;
  width: 70%;
}
.form-label {
  color: gray;
  font-size: 13px;
  text-align: justify;
  margin-top: 1%;
}
.row {
  margin-bottom: 0px;
}
.react-date-picker__wrapper {
  width: 200% !important;
  margin-top: 10% !important;
}
.step1-modal-body {
  padding-left: 10% !important;
  padding-bottom: 20% !important;
}
.step2-modal-body {
  padding-left: 10% !important;
  padding-bottom: 0% !important;
}
.modal-footer {
  padding-bottom: 10% !important;
}
.btn-info-solid-signup {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 25px;
  color: white;
  letter-spacing: 1px;
  line-height: 15px;
  border: 2px solid #00acee;
  border-radius: 40px;
  transition: all 0.3s ease 0s;
  background-color: #00acee;
  height: 40px;
  width: 300px;
}
.error {
  margin-left: -10px;
  color: red;
  font-weight: bold;
  font-size: 10px;
}
.registerModal {
  width: 50vw !important;
  max-width: 50vw !important;
}
.loginUserNameError {
  margin-left: 5%;
  color: red;
  font-weight: bold;
  font-size: 10px;
}
.twitterImage {
  margin-top: 22% !important;
  margin-bottom: 5% !important;
}
.joinTwitter {
  margin-top: 8%;
  margin-bottom: 3%;
}
.bolder {
  font-weight: bolder;
}
.colorP {
  color: #00acee;
  display: inline;
}
.next {
  margin-bottom: 10%;
  margin-top: 5%;
}
.createAccount {
  margin-bottom: 3%;
}
.margin {
  margin-top: 0%;
}
.marginTop20 {
  margin-top: 1%;
}
.marginTop2 {
  margin-top: 3%;
}
.signupMargin {
  margin-top: 10%;
  width: 100%;
}
.form-label-margin {
  color: gray;
  font-size: 13px;
  text-align: justify;
  margin-top: 0%;
}

*:focus {
  outline: 0 !important;
}
.registerModalHeader{
  flex-direction : row !important;
}

.error::before{
  content: "\200B";
}

.loginUserNameError::before{
  content: "\200B";
}