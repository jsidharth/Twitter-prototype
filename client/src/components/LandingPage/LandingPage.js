import React, { Component } from 'react';
import Modal from 'react-bootstrap/Modal';
import './LandingPage.css';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import DatePicker from 'react-date-picker';
import moment from 'moment';
import { connect } from 'react-redux';
import { userActions } from '../../js/actions/UserActions';
import { Redirect } from 'react-router';
import cookie from 'js-cookie';

class LandingPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showStep1: false,
      showStep2: false,
      dateOfBirth: '',
      name: '',
      email: '',
      date: '',
      password: '',
      nameError: '',
      emailError: '',
      dateError: '',
      passwordError: '',
      loginUserName: '',
      loginPassword: '',
      loginUserNameError: '',
      loginPasswordError: '',
    };
    this.showStep1Modal = this.showStep1Modal.bind(this);
    this.hideStep1Modal = this.hideStep1Modal.bind(this);
    this.showStep2Modal = this.showStep2Modal.bind(this);
    this.hideStep2Modal = this.hideStep2Modal.bind(this);
    this.nameChangeHandler = this.nameChangeHandler.bind(this);
  }
  dateChangeHandler = date => {
    this.setState({ date: date });
    if (date) {
      let month = date.getMonth() + 1;
      this.setState({ dateOfBirth: month + '/' + date.getDate() + '/' + date.getFullYear() });
    } else {
      this.setState({ dateOfBirth: '' });
    }
  };

  nameChangeHandler = e => {
    this.setState({ name: e.target.value });
  };
  emailChangeHandler = e => {
    this.setState({ email: e.target.value });
  };
  passwordChangeHandler = e => {
    this.setState({ password: e.target.value });
  };
  loginUserNameChangeHandler = e => {
    this.setState({ loginUserName: e.target.value });
  };
  loginPasswordChangeHandler = e => {
    this.setState({ loginPassword: e.target.value });
  };
  showStep1Modal = () => {
    this.setState({ showStep1: true, showStep2: false });
  };
  hideStep1Modal = () => {
    this.setState({ showStep1: false });
  };
  isPasswordValid = () => {
    console.log(this.state.password.length);
    if (this.state.password.length >= 6) {
      this.setState({ passwordError: '' });
      return true;
    } else {
      this.setState({ passwordError: '6 Characters minimum' });
      return false;
    }
  };
  isNameValid = () => {
    if (/^[a-zA-Z ]+$/.test(this.state.name)) {
      this.setState({ nameError: '' });
      return true;
    } else {
      this.setState({ nameError: 'Name needs to be all alphabets' });
      return false;
    }
  };
  isEmailValid = () => {
    if (/\S+@\S+\.\S+/.test(this.state.email)) {
      this.setState({ emailError: '' });
      return true;
    } else {
      this.setState({ emailError: 'Email format invalid' });
      return false;
    }
  };
  checkAgeAbove18 = () => {
    var dob = moment(this.state.date);
    var today = moment(new Date());
    var diffDuration = moment.duration(today.diff(dob));
    console.log();
    let ageInYears = diffDuration.years();
    if (ageInYears >= 18) {
      return true;
    } else {
      return false;
    }
  };
  isAgeValid = () => {
    if (this.state.dateOfBirth) {
      if (this.checkAgeAbove18()) {
        this.setState({ dateError: '' });
        return true;
      } else {
        this.setState({ dateError: 'You must be 18 years or above' });
        return false;
      }
    } else {
      this.setState({ dateError: '' });

      return true;
    }
  };
  showStep2Modal = () => {
    let nameValidity = this.isNameValid();
    let emailValidity = this.isEmailValid();
    let ageValidity = this.isAgeValid();
    let passwordValidity = this.isPasswordValid();
    if (
      this.state.name &&
      this.state.email &&
      this.state.dateOfBirth &&
      nameValidity &&
      emailValidity &&
      ageValidity &&
      passwordValidity
    ) {
      this.setState({ showStep2: true, showStep1: false });
    }
  };
  hideStep2Modal = () => {
    this.setState({ showStep2: false });
  };
  isLoginNameValid = () => {
    if (/\S+@\S+\.\S+/.test(this.state.loginUserName)) {
      this.setState({ loginUserNameError: '' });
      return true;
    } else {
      this.setState({ loginUserNameError: 'Wrong format of Email' });
      return false;
    }
  };
  login = () => {
    let userDetails = {
      email: this.state.loginUserName,
      password: this.state.loginPassword,
    };
    let loginNameValidity = this.isLoginNameValid();
    if (loginNameValidity && this.state.loginPassword) {
      this.props.login(userDetails);
    }
  };
  register = () => {
    let userDetails = {
      email: this.state.email,
      name: this.state.name,
      dob: this.state.dateOfBirth,
      password: this.state.password,
    };
    this.props.register(userDetails);
  };
  render() {
    let redirectVar = '';
    if (cookie.get('token')) {
      redirectVar = <Redirect to="/home" />;
    }
    return (
      <div>
        {redirectVar}
        <Row className="main1">
          <Col xs={6} className="home"></Col>
          <Col xs={6} className="">
            <Row>
              <Col md={4}>
                <Row>
                  <h6 className="login-label">Email</h6>
                </Row>
                <Row>
                  <input
                    className="login-username"
                    type="text"
                    value={this.state.loginUserName}
                    onChange={this.loginUserNameChangeHandler}
                  />
                </Row>
                <label class="loginUserNameError">{this.state.loginUserNameError}</label>
              </Col>
              <Col md={4}>
                <Row>
                  <h6 className="login-label">Password</h6>
                </Row>
                <Row>
                  <input
                    className="login-username"
                    type="password"
                    value={this.state.loginPassword}
                    onChange={this.loginPasswordChangeHandler}
                  />
                </Row>
                <label className="error">{this.state.loginPasswordError}</label>
              </Col>
              <Col md={2}>
                <button type="button" className=" btn-info-transparent-login" onClick={this.login}>
                  Log In
                </button>
              </Col>
              <Col md={1}></Col>
            </Row>
            <Row className="twitterImage">
              <Col md={3}></Col>
              <Col md={6}>
                <img
                  src="https://upload.wikimedia.org/wikipedia/fr/thumb/c/c8/Twitter_Bird.svg/944px-Twitter_Bird.svg.png"
                  width="40px"
                  height="40px"
                  alt="Main logo link to home"
                />{' '}
              </Col>
              <Col md={3}></Col>
            </Row>
            <Row className="">
              <Col md={3}></Col>
              <Col md={6}>
                <h2 className="bolder">See what's happening in the world right now.</h2>
              </Col>
              <Col md={3}></Col>
            </Row>
            <Row className="joinTwitter">
              <Col md={3}></Col>
              <Col md={6}>
                <h4 className="bolder">Join Twitter Today.</h4>
              </Col>
              <Col md={3}></Col>
            </Row>
            <Row className="">
              <Col md={3}></Col>
              <Col md={6}>
                <button type="button" className=" btn-info-solid" onClick={this.showStep1Modal}>
                  Sign Up
                </button>
              </Col>
              <Col md={3}></Col>
            </Row>
            <Row>
              <Col md={3}></Col>
              <Col md={6}>
                <button type="button" className=" btn-info-transparent">
                  Log In
                </button>
              </Col>
              <Col md={3}></Col>
            </Row>
          </Col>

          <>
            <Modal
              dialogClassName="registerModal"
              show={this.state.showStep1}
              onHide={this.hideStep1Modal}
              centered
            >
              <Modal.Header>
                <Col md={5}></Col>
                <Col md={3}>
                  <img
                    src="https://upload.wikimedia.org/wikipedia/fr/thumb/c/c8/Twitter_Bird.svg/944px-Twitter_Bird.svg.png"
                    width="25px"
                    height="25px"
                    alt="Main logo link to home"
                  />
                </Col>
                <Col md={2} />
                <Col md={2}>
                  <button
                    type="submit"
                    className="btn-info-solid-next"
                    onClick={this.showStep2Modal}
                  >
                    <p className="next">Next</p>
                  </button>
                </Col>
              </Modal.Header>
              <Modal.Body className="step1-modal-body">
                <Row className="createAccount">
                  <Col md={10}>
                    <h3 className="bolder">Create Your Account</h3>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row>
                  <Col md={1} />
                  <Col md={10}>
                    <Row>
                      <h6 className="">Name</h6>
                    </Row>
                    <Row>
                      <input
                        type="text"
                        value={this.state.name}
                        onChange={this.nameChangeHandler}
                      />
                    </Row>
                    <label className="error">{this.state.nameError}</label>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row>
                  <Col md={1}></Col>
                  <Col md={10}>
                    <Row className="margin">
                      <h6 className="">Email</h6>
                    </Row>
                    <Row>
                      <input
                        type="email"
                        value={this.state.email}
                        onChange={this.emailChangeHandler}
                      />
                    </Row>
                    <label className="error">{this.state.emailError}</label>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row>
                  <Col md={1}></Col>
                  <Col md={10}>
                    <Row className="margin">
                      <h6 className="">Password</h6>
                    </Row>
                    <Row>
                      <input
                        type="password"
                        value={this.state.password}
                        onChange={this.passwordChangeHandler}
                      />
                    </Row>
                    <label className="error">{this.state.passwordError}</label>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row>
                  <Col md={1} />
                  <Col md={10}>
                    <Row className="margin">
                      <h6 className="bolder">Date of Birth</h6>
                    </Row>
                    <Row>
                      <h6 className="form-label">
                        This should be the date of birth of the person creating this account,
                        whether its personal or business account. This won't be public unless you
                        change your profile settings.
                      </h6>
                    </Row>
                    <Row>
                      <DatePicker onChange={this.dateChangeHandler} value={this.state.date} />
                    </Row>
                    <label className="error">{this.state.dateError}</label>
                  </Col>
                  <Col md={1} />
                </Row>
              </Modal.Body>
            </Modal>
            <Modal
              dialogClassName="registerModal"
              show={this.state.showStep2}
              onHide={this.hideStep2Modal}
              centered
            >
              <Modal.Header>
                <Col md={2}>
                  <button
                    type="button"
                    className="btn-info-solid-next"
                    onClick={this.showStep1Modal}
                  >
                    <p className="next">Back</p>
                  </button>
                </Col>
                <Col md={2}></Col>
                <Col md={4}>
                  <h4 className="bolder">Step 2 of 2</h4>
                </Col>
                <Col md={4}></Col>
              </Modal.Header>
              <Modal.Body className="step2-modal-body">
                <Row className="marginTop20">
                  <Col md={10}>
                    <h3 className="bolder">Create Your Account</h3>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row>
                  <Col md={1} />
                  <Col md={10}>
                    <Row>
                      <h6 className="">Name</h6>
                    </Row>
                    <Row>
                      <input
                        type="text"
                        value={this.state.name}
                        readOnly
                        onClick={this.showStep1Modal}
                      />
                    </Row>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row className="marginTop2">
                  <Col md={1}></Col>
                  <Col md={10}>
                    <Row >
                      <h6 className="">Email</h6>
                    </Row>
                    <Row>
                      <input
                        type="text"
                        value={this.state.email}
                        readOnly
                        onClick={this.showStep1Modal}
                      />
                    </Row>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row>
                  <Col md={1}></Col>
                  <Col md={10}>
                    <Row className="marginTop2">
                      <h6 className="">Password</h6>
                    </Row>
                    <Row>
                      <input
                        type="password"
                        value={this.state.password}
                        onChange={this.passwordChangeHandler}
                        readOnly
                      />
                    </Row>
                  </Col>
                  <Col md={1} />
                </Row>
                <Row>
                  <Col md={1} />
                  <Col md={10}>
                    <Row className="marginTop2">
                      <h6 className="">Date of Birth</h6>
                    </Row>
                    <Row>
                      <input
                        type="text"
                        readOnly
                        value={this.state.dateOfBirth}
                        onClick={this.showStep1Modal}
                      />
                    </Row>
                  </Col>
                  <Col md={1} />
                </Row>
              </Modal.Body>
              <Modal.Footer>
                <Row>
                  <Row>
                    <Col md={2}></Col>
                    <Col md={8} className="marginLeft">
                      <h6 className="form-label">
                        By signing up, you agree to the
                        <p className="colorP"> Terms of Service</p> and
                        <p className="colorP"> Privacy Policy</p>, including
                        <p className="colorP"> Cookie Use</p>. Others will be able to find you by
                        email when provided.
                      </h6>
                    </Col>
                    <Col md={2}></Col>
                  </Row>
                  <Row>
                    <Col md={7}></Col>
                    <Col md={5} className="signupMargin">
                      <button className="btn-info-solid-signup" onClick={this.register}>
                        Sign Up
                      </button>
                    </Col>
                  </Row>
                </Row>
              </Modal.Footer>
            </Modal>
          </>
        </Row>
      </div>
    );
  }
}
const mapStateToProps = state => {
  return {
    userDetails: state.user.userDetails,
  };
};

const mapDispatchToProps = dispatch => ({
  register: data => dispatch(userActions.register(data)),
  login: data => dispatch(userActions.login(data)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LandingPage);
