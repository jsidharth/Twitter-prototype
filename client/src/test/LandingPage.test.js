// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import LandingPage from '../components/LandingPage/LandingPage';

configure({ adapter: new Adapter() });

it('render correctly the Landing Page component', () => {
  const LandingPageComponent = renderer.create(<LandingPage />).toJSON();
  expect(LandingPageComponent).toMatchSnapshot();
});
