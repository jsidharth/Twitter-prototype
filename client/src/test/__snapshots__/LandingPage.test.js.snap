// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly the Landing Page component 1`] = `
<div>
  
  <div
    className="main1 row"
  >
    <div
      className="home col-6"
    />
    <div
      className="col-6"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <div
            className="row"
          >
            <h6
              className="login-label"
            >
              Email
            </h6>
          </div>
          <div
            className="row"
          >
            <input
              className="login-username"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <label
            className="loginUserNameError"
          >
            
          </label>
        </div>
        <div
          className="col-md-4"
        >
          <div
            className="row"
          >
            <h6
              className="login-label"
            >
              Password
            </h6>
          </div>
          <div
            className="row"
          >
            <input
              className="login-username"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <label
            className="loginUserNameError"
          >
            
          </label>
        </div>
        <div
          className="col-md-2"
        >
          <button
            className=" info-transparent-login"
            onClick={[Function]}
            type="button"
          >
            Log In
          </button>
        </div>
      </div>
      <div
        className="twitterImage row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-6"
        >
          <img
            alt="Main logo link to home"
            height="40px"
            src="https://upload.wikimedia.org/wikipedia/fr/thumb/c/c8/Twitter_Bird.svg/944px-Twitter_Bird.svg.png"
            width="40px"
          />
           
        </div>
        <div
          className="col-md-3"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-6"
        >
          <h2
            className="bolder"
          >
            See what's happening in the world right now.
          </h2>
        </div>
        <div
          className="col-md-3"
        />
      </div>
      <div
        className="joinTwitter row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-6"
        >
          <h4
            className="bolder"
          >
            Join Twitter today.
          </h4>
        </div>
        <div
          className="col-md-3"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-6"
        >
          <button
            className=" btn-info-solid"
            onClick={[Function]}
            type="button"
          >
            Sign Up
          </button>
        </div>
        <div
          className="col-md-3"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-6"
        >
          <button
            className=" btn-info-transparent"
            type="button"
          >
            Log In
          </button>
        </div>
        <div
          className="col-md-3"
        />
      </div>
    </div>
  </div>
</div>
`;
